data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorker.java

```diff
@@ -102,12 +102,23 @@ public void parseS3Object(final S3ObjectReference s3ObjectReference, final Ackno
         }
     }
 
+    @Override
+    public Long getS3ObjectSize(S3ObjectReference s3ObjectReference) {
```

Comment:
Minor: It'd be nice to reduce some of the duplicate logic. I think the getSize method could be pushed into the `S3ObjectReference` class or a new helper rather than being an interface method of the workers. There'd need to be an overload unfortunately since the S3 clients are different, but generating the HeadObjectRequest could be shared.

Reply:
I looked for better ways to do this, cause it is duplicated for sure. The other option was to pass an instance variable to the `parseS3Object` method (which would eliminate the second call to get the object size that is made to actually process it), but that's pretty bad practice. 

I don't think we can add a getSize method to the `S3ObjectReference` though in the current state. We would have to pass the client to the `s3objectReference`, but in the case of SQS it's created by the SQS worker, so I'm not really sure how that would work

Reply:
The S3 notification already includes the object size. See https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html

It is available at `s3.object.size` in each record.

If you don't want to trust that, the `S3Service` can just do this.

The `S3ClientBuilderFactory` class is created in `S3Source` line 66. This is before the constructor for `S3Service`.

You can change that line to:

```
final S3Service s3Service = new S3Service(s3Handler, s3ClientBuilderFactory.getS3Client());
```

Reply:
Thanks for this info. I was looking for a way to see the object size without making a request to get it. Is there any reason I shouldn't trust the `s3.object.size`? If not I am fine with using that